src/Rx/Weno/TransmitProperties.php

```diff
@@ -0,0 +1,213 @@
+        //default is testing mode
+        $testing = isset($GLOBALS['weno_rx_enable_test']);
+        if ($testing) {
+            $mode = 'Y';
+        } else {
+            $mode = 'N';
+        }
+        $gender = $this->patient['sex'];
+        $heighDate = explode(" ", $this->vitals['date']);
+        if ($this->subscriber == 'self') {
+            $relationship = 'Y';
+        } else {
+            $relationship = 'N';
+        }
+        //create json array
+        $wenObj = [
+                            'UserEmail' => $this->provider_email['email'],
+                          'MD5Password' => md5($this->provider_pass),
+                           "LocationID" => $this->locid['weno_id'],
+                          "TestPatient" => $mode,
+                          'PatientType' => 'Human',
+                         'OrgPatientID' => $this->patient['pid'],
+                             'LastName' => $this->patient['lname'],
+                            'FirstName' => $this->patient['fname'],
+                           'MiddleName' => $this->patient['mname'],
+                               'Prefix' => 'NA',
+                               'Suffix' => 'NA',
+                               "Gender" => $gender[0],
+                          "DateOfBirth" => $this->patient['dob'],
+                         "AddressLine1" => $this->patient['street'],
+                         "AddressLine2" => "NA",
+                                 "City" => $this->patient['city'],
+                                "State" => $this->patient['state'],
+                           "PostalCode" => $this->patient['postal_code'],
+                          "CountryCode" => "US",
+                         "PrimaryPhone" => $this->patient['phone_cell'],
+                          "SupportsSMS" => "Y",
+                         "PatientEmail" => $this->patient['email'],
+                        "PatientHeight" => $this->vitals['height'],
+                        "PatientWeight" => $this->vitals['weight'],
+          "HeightWeightObservationDate" => $heighDate[0],
+        "ResponsiblePartySameAsPatient" => 'Y',
+                      "PatientLocation" => "Home",
+                 "PrimaryPharmacyNCPCP" => $this->ncpdp,
+             "AlternativePharmacyNCPCP" => $this->ncpdp
+        ];
+
+        return json_encode($wenObj);
+    }
+
+    /**
+     * @return mixed
+     */
+    public function getProviderEmail()
+    {
+        $provider_info = sqlQuery("select email from users where username=? ", [$_SESSION["authUser"]]);
+        if (empty($provider_info['email'])) {
+            echo xlt('Provider email address is missing');
+            exit;
+        } else {
+            return $provider_info;
+        }
+    }
+
+    /**
+     * @return mixed
+     */
+    public function getFacilityInfo()
+    {
+        $locid = sqlQuery("select name, street, city, state, postal_code, phone, fax, weno_id from facility where id = ?", [isset($_SESSION['facilityId'])]);
```

Comment:
this doesn't look right, that isset will either return a true or false

Reply:
done